raiden/network/transport/matrix/rtc/web_rtc.py

```diff
@@ -44,21 +46,81 @@ def schedule_task(
         self.coroutines.append(task)
         return task
 
-    async def wait_for_coroutines(self) -> None:
-        log.debug("Waiting for coroutines", coroutines=self.coroutines)
-        await asyncio.gather(*self.coroutines)
+    async def wait_for_coroutines(self, cancel: bool = True) -> None:
+
+        if cancel:
+            self.cancel_all_pending()
+
+        pending_coroutines = [
+            coroutine
+            for coroutine in self.coroutines
+            if not coroutine.done() and not coroutine.cancelled()
+        ]
+        log.debug("Waiting for coroutines", coroutines=pending_coroutines)
+
+        try:
+            await asyncio.gather(*pending_coroutines)
+        except CancelledError:
```

Comment:
It would probably be better to pass `return_exceptions=True`. Otherwise any exception will cause the gather to abort and not continue waiting on the remaining tasks.

From the [docs](https://docs.python.org/3.9/library/asyncio-task.html#asyncio.gather):
> If `return_exceptions` is `False` (default), the first raised exception is immediately propagated to the task that awaits on `gather()`. Other awaitables in the aws sequence won’t be cancelled and will continue to run.